src/services/adService.ts

```diff
@@ -2,14 +2,54 @@ import * as async from 'async';
 import * as moment from 'moment';
 import { IAdModel } from '../db/';
 import { IAd } from '../interfaces/';
-import { getAdByURL, getAll, markSeltAds, save, update } from '../repositories/adRepository';
-import { get, getByFilter, getSoldCarsNumber } from '../repositories/adRepository';
+
+import {
+  countSold,
+  get,
+  getAdByURL,
+  getAll,
+  getByFilter,
+  getByModelId,
+  getSold,
+  getSoldCarsNumber,
+  markSeltAds,
+  save,
+  update
+} from '../repositories/adRepository';
+
+
+
 import * as tempAdRepository from '../repositories/tempAdRepository';
 
 export const getAllAds = async () => {
   return await getAll();
 };
 
+export const countSoldWithFilter = async (model: string, bodytype: string) => {
+  const payload = {
+    bodyTypeId: bodytype,
+    isSold: true,
+    modelId: model
+  };
+  return await countSold(payload);
+};
+
+export const getAdByModelId = async (modelId: string) => {
+  return await getByModelId(modelId);
+};
+
+export const getSoldAd = async (
+  modelsBuffer: string[],
+  bodyTypeIdsBuffer: string[]
+): Promise<any> => {
```

Comment:
Why is it any, not boolean here?

Reply:
It should have <IAd> type `cause this method returns all sold ads with current mark and body type.Fixed